src/CartItemDiscount.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Gloudemans\Shoppingcart;
+
+class CartItemDiscount
+{
+    /**
+     * @var int|float
+     */
+    private $value;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $description;
+
+    /**
+     * CartItemDiscount constructor.
+     * @param int|float $value
+     * @param string $type
+     * @param string $description
+     */
+    public function __construct($value, $type = 'currency', $description = '')
+    {
+        if ($value < 0 || $value > 100) {
```

Comment:
What if I want 350 currency discount, this will throw exception even when its correct. Should check only if `$value < 0` and then check if type is `percentage` and then if value is greater than 100.